rrpproxy/rrpproxy.py

```diff
@@ -159,16 +160,36 @@ def query_zone_list(self):
         """Query list of activated zones in account"""
         return self.call('QueryZoneList')
 
-    def get_domain_list_from_account(self, index=0, time_between_calls_in_seconds=1):
+    def get_domain_list_from_account(self, **kwargs):
         """Returns list of domains registered in your account"""
-        domain_list = self.query_domain_list(first=index)
+        domain_list = self.get_last_updated_domains_with_details(retrieve_all=True, **kwargs)
+
+        return [domain['name'] for domain in domain_list]
+
+    def get_last_updated_domains_with_details(self, index=0, retrieve_all=False, time_between_calls_in_seconds=1,
+                                              **kwargs):
+        """
+        Returns list of domains registered in your account with domain name, status, update_date,
+        nameservers, admin_contact, tech_contact, billing_contact and owner_contact
+
+        :param int index: Start results from this item
+        :param bool retrieve_all: Setting this to true will return all domains in the account, otherwise
+        this function will return the last 1000 updated domains
+        :param int time_between_calls_in_seconds: Custom delay in seconds between calls
+        """
+        domain_list = self.query_domain_list(first=index, type='ALL', orderby='DOMAINUPDATEDDATE', order='DESC',
+                                             **kwargs)
```

Comment:
That kwargs are passed not tested. With `domain_list = self.query_domain_list(first=index, type='ALL', orderby='DOMAINUPDATEDDATE', order='DESC')` the tests still pass.